build/common/installer/scripts/tomlparser-common-agent-config.rb

```diff
@@ -62,12 +63,32 @@ def populateSettingValuesFromConfigMap(parsedConfig)
   end
 end
 
+def getKubernetesMetadataCacheTTLSeconds(parsedConfig)
+  begin
+    if !parsedConfig.nil? && !parsedConfig[:agent_settings].nil?
+      k8s_metadata_config = parsedConfig[:agent_settings][:k8s_metadata_config]
+      if !k8s_metadata_config.nil? && !k8s_metadata_config[:kube_meta_cache_ttl_secs].nil?
+        ttl_value = k8s_metadata_config[:kube_meta_cache_ttl_secs]
+        if ttl_value.is_a?(Integer) && ttl_value >= 0
+          @logEnableKubernetesMetadataCacheTTLSeconds = ttl_value
+          puts "config::INFO: Using config map value: kube_meta_cache_ttl_secs = #{@logEnableKubernetesMetadataCacheTTLSeconds}"
+        else
+          puts "config::WARN: Invalid kube_meta_cache_ttl_secs value: Must be an integer and >= 0"
+        end
+      end
+    end
+  rescue => errorStr
+    puts "config::error: Exception while reading config settings for logEnableKubernetesMetadataCacheTTLSeconds - #{errorStr}"
+  end
+end
+
 @configSchemaVersion = ENV["AZMON_AGENT_CFG_SCHEMA_VERSION"]
 puts "****************Start Config Processing********************"
 if !@configSchemaVersion.nil? && !@configSchemaVersion.empty? && @configSchemaVersion.strip.casecmp("v1") == 0 #note v1 is the only supported schema version , so hardcoding it
   configMapSettings = parseConfigMap
   if !configMapSettings.nil?
     populateSettingValuesFromConfigMap(configMapSettings)
+    getKubernetesMetadataCacheTTLSeconds(configMapSettings)
```

Comment:
You dont need this and code under populateSettingValuesFromConfigMap can be moved to populateSettingValuesFromConfigMap function.

Reply:
changed